cudax/include/cuda/experimental/__graph/path_builder.cuh

```diff
@@ -0,0 +1,207 @@
+  //! \param __stream The stream to use for the capture.
+  //! \param __capture_fn A function that will be called with the stream to capture the nodes to.
+  template <typename _Fn>
+  _CCCL_HOST_API void legacy_stream_capture(stream_ref __stream, _Fn&& __capture_fn)
+  {
+    _CCCL_TRY_CUDA_API(
+      ::cudaStreamBeginCaptureToGraph,
+      "Failed to begin stream capture",
+      __stream.get(),
+      __graph_,
+      __nodes_.data(),
+      nullptr,
+      __nodes_.size(),
+      cudaStreamCaptureModeGlobal);
+
+    __capture_fn(__stream.get());
+
+    cudaGraph_t __graph_out = nullptr;
+
+    cudaStreamCaptureStatus __capture_status;
+    const cudaGraphNode_t* __last_captured_node = nullptr;
+    size_t __num_nodes                          = 0;
+    _CCCL_TRY_CUDA_API(
+      ::cudaStreamGetCaptureInfo,
+      "Failed to get stream capture info",
+      __stream.get(),
+      &__capture_status,
+      nullptr,
+      nullptr,
+      &__last_captured_node,
+      &__num_nodes);
+    if (__capture_status != cudaStreamCaptureStatusActive)
+    {
+      __throw_cuda_error(cudaErrorInvalidValue, "Stream capture no longer active", "cudaStreamGetCaptureInfo");
+    }
+    _CCCL_TRY_CUDA_API(::cudaStreamEndCapture, "Failed to end stream capture", __stream.get(), &__graph_out);
+    assert(__graph_out == __graph_);
+    assert(__num_nodes == 1);
+    __nodes_.clear();
+    __nodes_.push_back(__last_captured_node[0]);
+  }
+#endif // _CCCL_CTK_AT_LEAST(12, 3)
+
+  //! \brief Clear the path builder and set the dependency node.
+  //! Used by most APIs that operate on a path builder to insert a new node into the path.
+  //! \param __node The node to set as the dependency node.
+  _CCCL_HOST_API void __clear_and_set_dependency_node(cudaGraphNode_t __node)
+  {
+    __nodes_.clear(); // Clear existing nodes
+    __nodes_.push_back(__node);
+  }
+
+  //! \brief Get the dependencies of the path builder.
+  //! \return A span of the dependencies of the path builder.
+  [[nodiscard]] _CCCL_TRIVIAL_HOST_API auto get_dependencies() const noexcept -> _CUDA_VSTD::span<const cudaGraphNode_t>
+  {
+    return _CUDA_VSTD::span(__nodes_.data(), __nodes_.size());
+  }
+
+  //! \brief Add the dependencies of another path builder to the current path builder.
+  //! \param __other The path builder to add dependencies from.
+  //! Named wait to match the stream/stream_ref wait function
+  _CCCL_HOST_API void wait(path_builder __other)
+  {
+    __nodes_.insert(__nodes_.end(), __other.__nodes_.begin(), __other.__nodes_.end());
+  }
+
+  //! \brief Add the dependencies of another path builder or single nodes to the current path builder.
+  //! \param __nodes The nodes or path builders to add to the path builder as dependencies.
+  _CCCL_TEMPLATE(typename... Nodes)
+  _CCCL_REQUIRES((((_CUDA_VSTD::is_same_v<_CUDA_VSTD::decay_t<Nodes>, graph_node_ref>)
+                   || _CUDA_VSTD::is_same_v<_CUDA_VSTD::decay_t<Nodes>, path_builder>)
+                  && ...))
```

Comment:
We should reuse the concept you defined above.

Also there are some compilers that choke on fold expression in requires expressions, especially in our C++17 emulation. 
```suggestion
  template<class... Nodes>
  static constexpr bool __all_dependencies = (graph_dependency<Nodes> && ...);

  _CCCL_TEMPLATE(class... Nodes)
  _CCCL_REQUIRES(__all_dependencies<Nodes...>)
```